### PR TITLE
[Federation]  Add integration testing for cluster addition

### LINKED ISSUE #45257: TestConfigMapController {sync}
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-test-go/5345/
Failed: TestConfigMapController {sync}

```
0x231a71d	k8s.io/kubernetes/federation/pkg/federation-controller/util/test.RegisterFakeWatch.func1+0x14d	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/federation/pkg/federation-controller/util/test/test_helper.go:159
0x17b0b9d	k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*controller).Run.func1+0x4d	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/controller.go:103
0x468489	sync.runtime_notifyListWait+0x119														/usr/local/go/src/runtime/sema.go:298
0x48c856	sync.(*Cond).Wait+0x96																/usr/local/go/src/sync/cond.go:57
0x17a57bd	k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*DeltaFIFO).Pop+0x10d									/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/delta_fifo.go:438
0x17a1a80	k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*controller).processLoop+0x70								/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/controller.go:147
0x17b1e31	k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*controller).(k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.processLoop)-fm+0x41	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/controller.go:121
0xa808af	k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1+0x6f								/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:97
0xa8001d	k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil+0xcd									/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:98
0xa7ff2a	k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Until+0x5a										/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:52
0x17a17cc	k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*controller).Run+0x34c									/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/controller.go:121
0x17ac02a	k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*Reflector).watchHandler+0x16ea	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector.go:350
0x17a9f30	k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch+0x11b0	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector.go:321
0x17b130a	k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*Reflector).RunUntil.func1+0x4a	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector.go:200
0xa808af	k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1+0x6f	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:97
0xa8001d	k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil+0xcd		/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:98
0xa7ff2a	k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Until+0x5a			/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:52
0x17b1795	k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch.func1+0x405	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector.go:268
0x22b8396	k8s.io/kubernetes/federation/pkg/federation-controller/util.(*DelayingDeliverer).run+0x636	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/federation/pkg/federation-controller/util/delaying_deliverer.go:122
0x22be7c7	k8s.io/kubernetes/federation/pkg/federation-controller/util.(*DelayingDeliverer).StartWithHandler.func1+0x157	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/federation/pkg/federation-controller/util/delaying_deliverer.go:174
0x23c41ad	k8s.io/kubernetes/federation/pkg/federation-controller/sync.(*FederationSyncController).Run.func1+0x4d	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/federation/pkg/federation-controller/sync/controller.go:217
0x561611	testing.(*T).Run+0x581		/usr/local/go/src/testing/testing.go:698
0x56671a	testing.runTests.func1+0xaa	/usr/local/go/src/testing/testing.go:882
0x561017	testing.tRunner+0x107		/usr/local/go/src/testing/testing.go:657
0x562cf0	testing.runTests+0x4e0		/usr/local/go/src/testing/testing.go:888
0x561c53	testing.(*M).Run+0x1c3		/usr/local/go/src/testing/testing.go:822
0x23c784f	main.main+0x20f			k8s.io/kubernetes/federation/pkg/federation-controller/sync/_test/_testmain.go:46
0x455ce2	runtime.main+0x1a2		/usr/local/go/src/runtime/proc.go:185
0x17c9b55	k8s.io/kubernetes/vendor/k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartEventWatcher.func1+0xc5	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/record/event.go:229
0x9b625d	k8s.io/kubernetes/vendor/github.com/golang/glog.(*loggingT).flushDaemon+0x9d	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/golang/glog/glog.go:879
0xa81c7c	k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/watch.(*Broadcaster).loop+0x8c	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/watch/mux.go:201
0x22be656	k8s.io/kubernetes/federation/pkg/federation-controller/util.StartBackoffGC.func1+0x126	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/federation/pkg/federation-controller/util/backoff.go:28
0x5fe550	runtime/pprof.writeRuntimeProfile+0xc0								/usr/local/go/src/runtime/pprof/pprof.go:632
0x5fe2e9	runtime/pprof.writeGoroutine+0xc9								/usr/local/go/src/runtime/pprof/pprof.go:594
0x5f90af	runtime/pprof.(*Profile).WriteTo+0x51f								/usr/local/go/src/runtime/pprof/pprof.go:302
0x23196b7	k8s.io/kubernetes/federation/pkg/federation-controller/util/test.GetObjectFromChan+0x177	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/federation/pkg/federation-controller/util/test/test_helper.go:310
0x23bc90c	k8s.io/kubernetes/federation/pkg/federation-controller/sync.GetConfigMapFromChan+0x3c		/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/federation/pkg/federation-controller/sync/configmap_controller_test.go:167
0x23bbe4a	k8s.io/kubernetes/federation/pkg/federation-controller/sync.TestConfigMapController+0x1faa	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/federation/pkg/federation-controller/sync/configmap_controller_test.go:143
0x561017	testing.tRunner+0x107										/usr/local/go/src/testing/testing.go:657
<autogenerated>:12: 
                        
	Error Trace:	configmap_controller_test.go:144
	
	Error:		Expected value not to be nil.
	
<autogenerated>:12: 
                        
	Error Trace:	configmap_controller_test.go:161
	
	Error:		Should be true
	
```

### PR DESCRIPTION
**Pull request by marun**
This PR adds integration testing of the sync controller for cluster addition.  This ensures coverage equivalency between the integration tests and the old controller unit tests, so those tests are removed by this PR.

Resolves #45257

cc: @kubernetes/sig-federation-pr-reviews

### CODE REVIEW

**Comment by madhusudancs** [comment]
/assign @perotinus 

(for review as well)

File: test/integration/federation/crud_test.go [hunk 1]
```diff
@@ -35,41 +37,57 @@ func TestFederationCRUD(t *testing.T) {
...
-			crudtester := framework.NewFederatedTypeCRUDTester(t, adapter, fedFixture.ClusterClients)
-			obj := adapter.NewTestObject(uuid.New())
-			updatedObj := crudtester.CheckCreate(obj)
-			crudtester.CheckDelete(updatedObj, orphanDependents)
+			updatedObj := crudTester.CheckCreate(obj)
+			crudTester.CheckDelete(updatedObj, orphanDependents)
 		})
 	}
+
+	t.Run("Resource should be propagated to newly added cluster", func(t *testing.T) {
```
Comment (perotinus) [inline test/integration/federation/crud_test.go]: nit: "to a newly"

**Review by perotinus — APPROVED** [review]
I like seeing this much red in the diff!

/lgtm

File: test/integration/federation/crud_test.go (hunk 1 above)
Comment (marun) [inline test/integration/federation/crud_test.go]: done

**Comment by marun** [comment]
@k8s-bot kops aws e2e test this
@k8s-bot verify test this

**Comment by marun** [comment]
rebased

**Comment by marun** [comment]
Had to manually edit the bazel file, building from scratch with hack/update-bazel.sh was stripping package/licenses/load.

cc: @mikedanese

**Comment by marun** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by marun** [comment]
@csbell PTAL

**Pull request merged**